### PR TITLE
Tweak: add more transition animation supports on Explore feed cards

### PR DESCRIPTION
**Pull request by cooltey**
https://phabricator.wikimedia.org/T263259#6579105

This PR also updates the "On this day" card design regarding Robin's comment: https://phabricator.wikimedia.org/T261673#6582483

### CODE REVIEW

File: app/src/main/java/org/wikipedia/feed/view/ListCardItemView.java [hunk 1]
```diff
@@ -93,7 +94,7 @@ public ListCardItemView(Context context) {
 
     @OnClick void onClick(View view) {
         if (callback != null && entry != null && card != null) {
-            callback.onSelectPage(card, entry);
+            callback.onSelectPage(card, entry, TransitionUtil.getSharedElements(getContext(), titleView, subtitleView, imageView));
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/feed/view/ListCardItemView.java:97]: Are we sure we want the `title` and `subtitle` as shared elements here? They are a different font and different size, and don't really look smooth when transitioning.

File: app/src/main/java/org/wikipedia/feed/view/CardLargeHeaderView.java [hunk 2]
```diff
@@ -124,4 +126,8 @@ private void setGradientDrawableBackground(@ColorInt int leftColor, @ColorInt in
         gradientDrawable.setCornerRadius(getResources().getDimension(R.dimen.wiki_card_radius));
         borderContainer.setBackground(gradientDrawable);
     }
+
+    public Pair<View, String>[] getSharedElements() {
+        return TransitionUtil.getSharedElements(getContext(), titleView, subtitleView, imageView);
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/feed/view/CardLargeHeaderView.java:131]: The transition for this looks pretty good, except for RTL articles. Is there any way to fix it?

**Review by dbrant — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/feed/view/ListCardItemView.java (hunk 1 above)
Comment (cooltey) [reply app/src/main/java/org/wikipedia/feed/view/ListCardItemView.java]: That's a good question and I have removed them. Also I'll leave a comment on the ticket.

File: app/src/main/java/org/wikipedia/feed/view/CardLargeHeaderView.java (hunk 2 above)
Comment (cooltey) [reply app/src/main/java/org/wikipedia/feed/view/CardLargeHeaderView.java]: Good catch! Done.

**Pull request merged**